### PR TITLE
[MM-60303] Changing status in mobile app doesn't change your status in the webapp

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
- Renamed `addVisibleUsersInCurrentChannelToStatusPoll` to `addVisibleUsersInCurrentChannelAndSelfToStatusPoll`
- Added `currentUserId` to function `addVisibleUsersInCurrentChannelAndSelfToStatusPoll`

#### Ticket Link
Jira https://mattermost.atlassian.net/browse/MM-60303

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by devinbinnie — APPROVED** [review]
Code looks good. As a sanity check can we confirm that this update doesn't happen for automatic status updates?

File: webapp/channels/src/components/channel_layout/channel_controller.tsx [hunk 1]
```diff
@@ -51,11 +51,13 @@ export default function ChannelController(props: Props) {
...
     }, []);
 
+    // Starts a regular interval to fetch statuses of users.
+    // see function "addVisibleUsersInCurrentChannelAndSelfToStatusPoll" for more details on which user's statuses are fetched.
     useEffect(() => {
         let loadStatusesIntervalId: NodeJS.Timeout;
         if (enabledUserStatuses) {
             loadStatusesIntervalId = setInterval(() => {
-                dispatch(addVisibleUsersInCurrentChannelToStatusPoll());
+                dispatch(addVisibleUsersInCurrentChannelAndSelfToStatusPoll());
```
Comment (hmhealey) [inline webapp/channels/src/components/channel_layout/channel_controller.tsx:60]: As per your comment in MM, did you want to also rename "Poll" to "Pool" here?

**Review by hmhealey — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
> Code looks good. As a sanity check can we confirm that this update doesn't happen for automatic status updates?

Can you clarify what's automatic status updates? @devinbinnie

**Comment by devinbinnie** [comment]
> > Code looks good. As a sanity check can we confirm that this update doesn't happen for automatic status updates?
> 
> Can you clarify what's automatic status updates? @devinbinnie

When the app sets you to Online/Away automatically after a period of inactivity. The Desktop App does it after 5 minutes.

**Comment by M-ZubairAhmed** [comment]
> > > Code looks good. As a sanity check can we confirm that this update doesn't happen for automatic status updates?
> > 
> > 
> > Can you clarify what's automatic status updates? @devinbinnie
> 
> When the app sets you to Online/Away automatically after a period of inactivity. The Desktop App does it after 5 minutes.

Is it 5 mins? Am i looking at the correct code below?

https://github.com/mattermost/desktop/blob/42a0bc4759fb96d214220a085542fa1239a80bce/src/main/UserActivityMonitor.ts#L35

File: webapp/channels/src/components/channel_layout/channel_controller.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/channel_layout/channel_controller.tsx:60]: No i thought about it a bit more term "polling" refers to the process of periodically querying or fetching data from a source.

**Review by yasserfaraazkhan — APPROVED** [review]
Changing status on mobile will show the same status on browser and desktop after a minute.

**Pull request merged**